code/client/src/components/ThesisPage.jsx

```diff
@@ -139,8 +150,15 @@ function ThesisPage(props) {
                 </Link>
               )}
 
+              {/* Delete button (visible only for the supervisor) */}
+              {props.user.role === 'teacher' && (
+                <Button variant="outline-danger" className="mt-3 ms-2" onClick={handleDeleteProposal}>
+                  Delete Proposal
```

Comment:
You should do the check at this stage and hide the button if user is not supervisor because the co-supervisor doesn't need to see the delete proposal button

Reply:
fixed